lib/puppet/type/archive.rb

```diff
@@ -160,12 +160,16 @@ def should_to_s(value)
   end
 
   autorequire(:file) do
+    autos = []
     [
       Pathname.new(self[:path]).parent.to_s,
       self[:extract_path],
       '/root/.aws/config',
       '/root/.aws/credentials',
-    ]
+    ].each do |file| 
+      autos << file unless file.nil?
+    end
+    autos
```

Comment:
Please just do .compact on the array. Also we might consider only adding /root/.aws on posix platforms.